### ISSUE TITLE
`User` changes take significant time to propagate between threads

### ISSUE THREAD

**Issue by MobileVet** [parse-community/parse-server#7313]
### New Issue Checklist

- [X ] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [X ] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [X ] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [X ] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
Making a change to a `User` in one thread, does not show up immediately in another worker thread.  For instance, if you create a super simple cloud function (see below) that changes a property on the `User` object and another cloud function that reads back that value (after having it hydrated from the _request_)... even if you `await` the change function, if you call the **readUser** function immediately after you will get the original value back.

If you add an async wait and THEN call the **readUser** function, the value will be updated properly.  Even running locally the wait must be several **seconds** to be effective.

### Steps to reproduce
- Spawn **2 or more** instances of your server using Throng or similar Node clustering module
- Call **updateUser** function from client w/ `await`
- Call **fetchUser** function from client
- Notice that the updated value is NOT correct

### Actual Outcome
- Original value of `User.testVal`

### Expected Outcome
- Updated value of `User.testVal`

### Failing Test Case / Pull Request
Will look at this... not sure if test env has ability to utilize clustering.

- [ ] 🤩 I submitted a PR with a fix and a test case.
- [ ] 🧐 I submitted a PR with a failing test case.

###  Environment
**Server**
- Parse Server version: 4.5.0
- Operating system: macOS 10.15.7
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): local

**Database**
- System (MongoDB or Postgres): MongoDB
- Database version: 4.4.4
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): MongoDB Atlas

**Client**
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): JavaScript
- SDK version: 2.19.0

### Functions
```
Parse.Cloud.define('updateUser', async (req: Parse.Cloud.FunctionRequest) => {
    const user = req.user
    if (!user) {
        throw 'no user'
    }
    const token = user.getSessionToken()
    const updatedThing = await user.save({ testVal: String(new Date().getTime()) }, { sessionToken: token })
    console.log('updatedThing:', JSON.stringify(updatedThing.toJSON(), null, 4))
    console.log('pid:', JSON.stringify(process.pid, null, 4))
    console.log('testVal (after updating):', user.get('testVal'))
})
Parse.Cloud.define('fetchUser', async (req: Parse.Cloud.FunctionRequest) => {
    const user = req.user
    if (!user) {
        throw 'no user'
    }
    console.log('pid:', JSON.stringify(process.pid, null, 4))
    console.log('testVal:', user.get('testVal'))
})
```

### Client Test function
```
async clickTestThign () {
    await this.fetchUserTest()
    await this.updateUserTest()
    this.fetchUserTest()
    this.fetchUserTest()
    this.fetchUserTest()
    this.fetchUserTest()
    this.fetchUserTest()
    await new Promise<void>(res => setTimeout(async () => {
        this.fetchUserTest()
        this.fetchUserTest()
        this.fetchUserTest()
        this.fetchUserTest()
        res()
    }, 2000))
}
```

**Comment by mtrezza** [comment]
Thanks for reporting.

Would you open a PR with a failing test case?

**Comment by MobileVet** [comment]
@mtrezza Will take a look.  I didn't realize the cluster configuration was possible with Parse Server setup.  We have been using Throng outside of our PS setup.

**Comment by mtrezza** [comment]
Yes, not only is it possible, it should be conveniently configurable with a single option.
